content/en/blog/_posts/2021-04-16-volume-health-monitoring-alpha.md

```diff
@@ -0,0 +1,95 @@
+---
+layout: blog
+title: "Volume Health Monitoring goes Alpha again"
+date: 2021-04-16
+slug: volume-health-monitoring-go-alpha
+---
+
+**Author:** Xing Yang (VMware)
+
+A second alpha version of CSI Volume Health Monitoring feature was released with Kubernetes 1.21. It was introduced as an alpha feature in Kubernetes 1.19. In Kubernetes 1.21, a second alpha was done with some design changes.
+
+## Why add Volume Health Monitoring to Kubernetes?
+
+Without Volume Health Monitoring, Kubernetes has no knowledge of what happened to the underlying volumes on the storage system after a PVC is provisioned and used by a Pod. Many things could happen to the underlying storage system after a volume is provisioned in Kubernetes. For example, the volume could be deleted by accident outside of Kubernetes, the disk that the volume resides on could fail, the volume could be out of capacity, the disk may be degrading which affects its performance, and so on. Even when the volume is mounted on a pod and used by an application, there could be problems later on such as read/write I/O errors, file system corruption, accidental unmounting of the volume outside of Kubernetes, etc. It is very hard to debug and detect root causes when something happened like this.
+
+With Volume Health Monitoring, Kubernetes could communicate with the CSI driver to retrieve any errors detected by the underlying storage system. Kubernetes can report an event about this PVC so that user can inquire this information and decide how to handle them. For example, if the volume is out of capacity, user can request a volume expansion to get more space. Volume health monitoring is a very important tool that would be beneficial to Kubernetes users.
+
+## What is Volume Health Monitoring?
+
+CSI Volume Health Monitoring allows CSI Drivers to detect abnormal volume conditions from the underlying storage systems and report them as events on PVCs or Pods.
+
+The Kubernetes components that monitor the volumes and report events with volume health information include the following:
+
+* In addition to gathering existing volume stats, Kubelet also watches volume health of the PVCs on that node. If a PVC has an abnormal health condition, an event will to reported on the pod object that is using the PVC. If multiple pods are using the same PVC, events will be reported on multiple pods.
```

Comment:
```suggestion
* Kubelet, in addition to gathering the existing volume stats will watch the volume health of the PVCs on that node. If a PVC has an abnormal health condition, an event will be reported on the pod object using the PVC. If multiple pods are using the same PVC, events will be reported on all pods using that PVC.
```
Slight rewording to state the component being modified earlier.

Reply:
Done